modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/notifications/SoapServiceEmailNotificationTestCase.java

```diff
@@ -0,0 +1,351 @@
+import org.wso2.carbon.identity.user.profile.stub.types.UserFieldDTO;
+import org.wso2.carbon.identity.user.profile.stub.types.UserProfileDTO;
+import org.wso2.carbon.integration.common.utils.exceptions.AutomationUtilException;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.es.utils.EmailUtil;
+import org.wso2.greg.integration.common.clients.UserProfileMgtServiceClient;
+import org.wso2.greg.integration.common.utils.GREGIntegrationBaseTest;
+import org.wso2.greg.integration.common.utils.GenericRestClient;
+
+import javax.ws.rs.core.MediaType;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * This test class can be used to check the email notification functionality
+ */
+public class SoapServiceEmailNotificationTestCase extends GREGIntegrationBaseTest {
+
+    private TestUserMode userMode;
+    String jSessionId;
+    private UserProfileMgtServiceClient userProfileMgtClient;
+    private File axis2File;
+    private String publisherUrl;
+    private String resourcePath;
+    private String assetId;
+    private String cookieHeader;
+    private GenericRestClient genericRestClient;
+    private Map<String, String> queryParamMap;
+    private Map<String, String> headerMap;
+    private String loginURL;
+    private String emailAddress;
+    private EmailUtil emailUtil;
+    boolean isNotificationMailAvailable;
+
+
+    @Factory(dataProvider = "userModeProvider")
+    public SoapServiceEmailNotificationTestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+
+        super.init(userMode);
+        loginURL = UrlGenerationUtil.getLoginURL(automationContext.getInstance());
+        emailAddress = "gregtestes@gmail.com";
+        genericRestClient = new GenericRestClient();
+        queryParamMap = new HashMap<>();
+        headerMap = new HashMap<>();
+        resourcePath =
+                FrameworkPathUtil.getSystemResourceLocation() + "artifacts" + File.separator + "GREG" + File.separator;
+        publisherUrl = automationContext.getContextUrls().getSecureServiceUrl().replace("services", "publisher/apis");
+        userProfileMgtClient = new UserProfileMgtServiceClient(backendURL, sessionCookie);
+        axis2File = new File(
+                TestConfigurationProvider.getResourceLocation("GREG") + File.separator + "axis2" + File.separator
+                        + "axis2.xml");
+
+        updateProfileAndEnableEmailConfiguration();
+        setTestEnvironment();
+
+    }
+
+    private void updateProfileAndEnableEmailConfiguration()
```

Comment:
Removed to EmailUtil class , because used by all email subscription test classes.
Refer PR #343